cmd/stat.go

```diff
@@ -0,0 +1,222 @@
+	case runtime.GOOS == "windows":
+		c.URL.Path = strings.Replace(c.URL.Path, "/", "\\", -1)
+		c.URL.Path = strings.TrimSuffix(c.URL.Path, "\\")
+	default:
+		c.URL.Path = strings.TrimSuffix(c.URL.Path, "/")
+	}
+	if c.Type.IsDir() {
+		switch {
+		// for windows make sure to print in 'windows' specific style.
+		case runtime.GOOS == "windows":
+			return fmt.Sprintf("%s\\", c.URL.Path)
+		default:
+			return fmt.Sprintf("%s/", c.URL.Path)
+		}
+	}
+	return c.URL.Path
+}
+
+// doStat - list all entities inside a folder.
+func doStat(clnt Client, isRecursive bool, targetURL string) error {
+	if !isRecursive || !isAliasURLDir(targetURL) {
+		isIncomplete := false
+		isFetchMeta := true
+		if st, err := clnt.Stat(isIncomplete, isFetchMeta); err != nil {
+			switch err.ToGoError().(type) {
+			default:
+				fatalIf(err.Trace(targetURL), "Unable to initialize `"+targetURL+"`.")
+			}
+		} else {
+			printStats([]statMessage{parseStat(st)}...)
+			return nil
+		}
+	}
+	if !strings.HasSuffix(targetURL, string(clnt.GetURL().Separator)) {
+		targetURL = targetURL + string(clnt.GetURL().Separator)
+	}
+	clnt, err := newClient(targetURL)
+	fatalIf(err.Trace(targetURL), "Unable to initialize target `"+targetURL+"`.")
+
+	prefixPath := clnt.GetURL().Path
+	separator := string(clnt.GetURL().Separator)
+	if !strings.HasSuffix(prefixPath, separator) {
+		prefixPath = prefixPath[:strings.LastIndex(prefixPath, separator)+1]
+	}
+	var cErr error
+	isIncomplete := false
+	var stats []statMessage
+
+	for content := range clnt.List(isRecursive, isIncomplete, DirNone) {
+		if content.Err != nil {
+			switch content.Err.ToGoError().(type) {
+			// handle this specifically for filesystem related errors.
+			case BrokenSymlink:
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list broken link.")
+				continue
+			case TooManyLevelsSymlink:
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list too many levels link.")
+				continue
+			case PathNotFound:
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list folder.")
+				continue
+			case PathInsufficientPermission:
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list folder.")
+				continue
+			case ObjectOnGlacier:
+				errorIf(content.Err.Trace(clnt.GetURL().String()), "")
+				continue
+			}
+			errorIf(content.Err.Trace(clnt.GetURL().String()), "Unable to list folder.")
+			cErr = exitStatus(globalErrorExitStatus) // Set the exit status.
+			continue
+		}
+		// Convert any os specific delimiters to "/".
+		contentURL := filepath.ToSlash(content.URL.Path)
+		prefixPath = filepath.ToSlash(prefixPath)
+		// Trim prefix path from the content path.
+		contentURL = strings.TrimPrefix(contentURL, prefixPath)
+		content.URL.Path = contentURL
+		//parsedContent := parseStat(content)
+		_, stat, err := url2StatWithMetadata(targetURL+getKey(content), true)
+		if err != nil {
+			continue
+		}
+		stats = append(stats, parseStat(stat))
```

Comment:
We cannot build the list like this we should simply printStat() one by one as the list occurs.

Reply:
good point , this increased the speed of stat command.